x-pack/filebeat/module/iptables/log/ingest/pipeline.json

```diff
@@ -0,0 +1,332 @@
+                        "IPTABLES_IP_PAYLOAD": "%{IPTABLES_PROTOCOL}( %{IPTABLES_PORT_PAIR})?( (%{IPTABLES_TCP_DETAILS}|%{IPTABLES_UDP_DETAILS}|%{IPTABLES_ICMP_DETAILS}|%{IPTABLES_INCOMPLETE_PACKET}))?",
+                        "IPTABLES_IP_FRAGFLAG": "((?<= )(CE|DF|MF))*",
+                        "IPTABLES_IP_START": "SRC=%{IPV4:source.ip} DST=%{IPV4:destination.ip} LEN=%{UNSIGNED_INT:iptables.length} TOS=0x%{BASE16NUM:iptables.ip_tos} PREC=0x%{BASE16NUM:iptables.ip_precedence_bits} TTL=%{UNSIGNED_INT:iptables.ttl} ID=%{UNSIGNED_INT:iptables.id}(?: %{IPTABLES_IP_FRAGFLAG:iptables.fragment_flags})?(?: FRAG: %{UNSIGNED_INT:iptables.fragment_offset})?",
+                        "IPTABLES_IP": "%{IPTABLES_IP_START} %{IPTABLES_IP_PAYLOAD}",
+                        "IPTABLES_IPV6": "SRC=%{IPV6:source.ip} DST=%{IPV6:destination.ip} LEN=%{UNSIGNED_INT:iptables.length} TC=%{UNSIGNED_INT:iptables.ip_tos} HOPLIMIT=%{UNSIGNED_INT:iptables.ttl} FLOWLBL=%{UNSIGNED_INT:iptables.flow_label} %{IPTABLES_IP_PAYLOAD}",
+                        "IPTABLES": "%{IPTABLES_ETHERNET} (:?%{IPTABLES_IP}|%{IPTABLES_IPV6})",
+                        "UBIQUITI_FIELD": "[^-\\]]*",
+                        "UBIQUITI_RULESET_NAME": "[^\\]]*",
+                        "UBIQUITI_LABEL": "%{UBIQUITI_RULESET_NAME:iptables.ubiquiti.rule_set}-%{UBIQUITI_FIELD:iptables.ubiquiti.rule_number}-%{UBIQUITI_FIELD:event.outcome}"
+                    }
+                }
+            },
+            {
+                "rename": {
+                    "field": "message",
+                    "target_field": "log.original"
+                }
+            },
+            {
+                "grok": {
+                    "field": "iptables.ubiquiti.rule_set",
+                    "ignore_missing": true,
+                    "ignore_failure": true,
+                    "patterns": [
+                        "%{UBIQUITI_FIELD:iptables.ubiquiti.input_interface}-%{UBIQUITI_FIELD:iptables.ubiquiti.output_interface}"
+                    ],
+                    "pattern_definitions": {
+                        "UBIQUITI_FIELD": "[^-]*"
+                    }
+                }
+            },
+            {
+                "date": {
+                    "field": "iptables.raw_date",
+                    "ignore_failure": true,
+                    {< if .convert_timezone >}"timezone": "{{ beat.timezone }}",{< end >}
+                    "formats": [
+                        "MMM  d HH:mm:ss",
+                        "MMM dd HH:mm:ss"
+                    ]
+                }
+            },
+            {
+                "remove": {
+                    "field": "iptables.raw_date",
+                    "ignore_missing": true
+                }
+            },
+
+            {
+                "convert": {
+                    "field": "source.port",
+                    "type": "integer",
+                    "ignore_missing": true
+                }
```

Comment:
Could you cast directly in the grok patterns with the [`%{SYNTAX:SEMANTIC:TYPE}` syntax](https://www.elastic.co/guide/en/elasticsearch/reference/6.5/grok-processor.html#grok-basics) instead of using convert?